calyx-py/calyx/builder.py

```diff
@@ -126,10 +134,39 @@ def output(self, name: str, size: int) -> ExprBuilder:
         self.component.outputs.append(ast.PortDef(ast.CompVar(name), size))
         return self.this()[name]
 
+    def output_with_attributes(self, name: str, size: int, attribute_literals: List[Union[str, Tuple[str, int]]]) -> ExprBuilder:
```

Comment:
Maybe I'm missing something obvious here, but could we make this slightly simpler by getting rid of the `*_with_attributes` versions of the methods and just adding an optional (keyword) argument to the existing `input` and `output` methods? Like, the signature would look like:

```py
def output(self, name: str, size: int, attrs: Attrs = []) -> ExprBuilder:
```

…so by default, there are not attributes, but people can supply them optionally if they want.